ns3_adapter/src/ns3_client.cpp

```diff
@@ -113,11 +121,48 @@ void NS3Client::close() {
     work_.reset();
     io_->stop();
     io_thread_->join();
-    udp_listener_->stop();
-    udp_out_socket_.reset();
+    udp_v2x_listener_->stop();
+    udp_time_listener_->stop();
+    udp_out_v2x_msg_broadcasting_socket_.reset();
+    udp_out_registration_socket_.reset();
     onDisconnect();
 }
 
+void NS3Client::process_time(const std::shared_ptr<const std::vector<uint8_t>>& data)
+{
+    const std::vector<uint8_t> vec = *data;
+    ROS_DEBUG_STREAM("process_time data received of size: " << vec.size());
+
+    std::string json_string(vec.begin(), vec.end());
+    ROS_DEBUG_STREAM("process_time data: " << json_string);
+
+    // JSON
+    rapidjson::Document obj;
+    std::string timestep_member_name = "timestep";
+    obj.Parse(json_string);
+    if (obj.HasParseError())
+    {
+        throw std::runtime_error("Message JSON is misformatted. JSON parsing failed! Please check process_time data: " + json_string);
+    }
+
+    std::optional<unsigned long> result = std::nullopt;
```

Comment:
This does not need to be optional. We should make this a regular variable

Reply:
maybe your cache is not updated or you are not looking at the last commit. the optional field is not there.